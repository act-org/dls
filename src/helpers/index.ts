/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

export * from './cssRadius';
export * from './getErrorMessage';
export * from './makeShadow';
export * from './mergeClasses';
export * from './oxfordCommaJoin';
export * from './px';
export * from './pxToNumber';
export * from './search';
export * from './sort';
export * from './styled';
export * from './types';
