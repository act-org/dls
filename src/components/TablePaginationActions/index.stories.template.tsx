/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import { Story } from '@storybook/react/types-6-0';
import { useState } from 'react';

import { Playground } from '@actinc/dls/helpers/playground';

import { TablePaginationActions, TablePaginationActionsProps } from '.';

export const Template: Story<TablePaginationActionsProps> = args => {
  const [page, setPage] = useState(0);

  return (
    <TablePaginationActions
      {...args}
      onPageChange={(_, p): void => {
        setPage(p);
      }}
      page={page}
    />
  );
};
Template.args = {};

export const argTypes = Playground(
  {
    count: {},
    disabled: {},
    noun: {},
    onPageChange: {},
    page: {},
    rowsPerPage: {},
    tooltipPlacement: {},
    variant: {},
  },
  TablePaginationActions,
);
