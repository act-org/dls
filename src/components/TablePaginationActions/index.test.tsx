/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import { noop } from 'lodash';

import { standard } from '~/helpers/test';

import TablePaginationActions from '.';

describe('TablePaginationActions', () => {
  const Component = (
    <TablePaginationActions
      count={1000}
      onPageChange={noop}
      page={1}
      rowsPerPage={100}
    />
  );

  standard(Component);
});
