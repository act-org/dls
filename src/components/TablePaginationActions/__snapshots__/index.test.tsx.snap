// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TablePaginationActions ACT theme matches the snapshot 1`] = `
<div>
  <div
    class="css-14q6vlk"
  >
    <span
      aria-label="First Page"
      class=""
      color="primary"
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="First Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-3vwr36-MuiButtonBase-root-MuiIconButton-root"
        color="primary"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-d1s5pq-MuiSvgIcon-root"
          data-testid="PageFirstIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18.41,16.59L13.82,12L18.41,7.41L17,6L11,12L17,18L18.41,16.59M6,6H8V18H6V6Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Previous Page"
      class=""
      color="primary"
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Previous Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-3vwr36-MuiButtonBase-root-MuiIconButton-root"
        color="primary"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-d1s5pq-MuiSvgIcon-root"
          data-testid="ChevronLeftIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
          />
        </svg>
      </button>
    </span>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-oxolt7-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-dpynhu-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-controls=":r3:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Select"
            aria-labelledby=":r2:"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-10ortyn-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r2:"
            role="combobox"
            tabindex="0"
          >
            <span>
              Page 1
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
            tabindex="-1"
            value="0"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-9wviix-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-w4cd9x"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <span
      aria-label="Next Page"
      class=""
      color="primary"
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Next Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-3vwr36-MuiButtonBase-root-MuiIconButton-root"
        color="primary"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-d1s5pq-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Last Page"
      class=""
      color="primary"
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Last Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-3vwr36-MuiButtonBase-root-MuiIconButton-root"
        color="primary"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-d1s5pq-MuiSvgIcon-root"
          data-testid="PageLastIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5.59,7.41L10.18,12L5.59,16.59L7,18L13,12L7,6L5.59,7.41M16,6H18V18H16V6Z"
          />
        </svg>
      </button>
    </span>
  </div>
</div>
`;

exports[`TablePaginationActions ACT_ET theme matches the snapshot 1`] = `
<div>
  <div
    class="css-14q6vlk"
  >
    <span
      aria-label="First Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="First Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1tpud1o-MuiSvgIcon-root"
          data-testid="PageFirstIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18.41,16.59L13.82,12L18.41,7.41L17,6L11,12L17,18L18.41,16.59M6,6H8V18H6V6Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Previous Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Previous Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1tpud1o-MuiSvgIcon-root"
          data-testid="ChevronLeftIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
          />
        </svg>
      </button>
    </span>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-q2rbby-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-by28yo-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-controls=":r9:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Select"
            aria-labelledby=":r8:"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-8ntq6y-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r8:"
            role="combobox"
            tabindex="0"
          >
            <span>
              Page 1
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
            tabindex="-1"
            value="0"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1iw9ayg-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-w4cd9x"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <span
      aria-label="Next Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Next Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1tpud1o-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Last Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Last Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1tpud1o-MuiSvgIcon-root"
          data-testid="PageLastIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5.59,7.41L10.18,12L5.59,16.59L7,18L13,12L7,6L5.59,7.41M16,6H18V18H16V6Z"
          />
        </svg>
      </button>
    </span>
  </div>
</div>
`;

exports[`TablePaginationActions ENCOURA theme matches the snapshot 1`] = `
<div>
  <div
    class="css-14q6vlk"
  >
    <span
      aria-label="First Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="First Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-75t6cl-MuiSvgIcon-root"
          data-testid="PageFirstIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18.41,16.59L13.82,12L18.41,7.41L17,6L11,12L17,18L18.41,16.59M6,6H8V18H6V6Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Previous Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Previous Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-75t6cl-MuiSvgIcon-root"
          data-testid="ChevronLeftIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
          />
        </svg>
      </button>
    </span>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-esfs78-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-formControl MuiInputBase-sizeSmall css-s1ym9j-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-controls=":rf:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Select"
            aria-labelledby=":re:"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-qyjbs4-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id=":re:"
            role="combobox"
            tabindex="0"
          >
            <span>
              Page 1
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
            tabindex="-1"
            value="0"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1p4sevb-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ChevronDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-w4cd9x"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <span
      aria-label="Next Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Next Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-75t6cl-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Last Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Last Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-75t6cl-MuiSvgIcon-root"
          data-testid="PageLastIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5.59,7.41L10.18,12L5.59,16.59L7,18L13,12L7,6L5.59,7.41M16,6H18V18H16V6Z"
          />
        </svg>
      </button>
    </span>
  </div>
</div>
`;

exports[`TablePaginationActions ENCOURA_CLASSIC theme matches the snapshot 1`] = `
<div>
  <div
    class="css-14q6vlk"
  >
    <span
      aria-label="First Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="First Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-75t6cl-MuiSvgIcon-root"
          data-testid="PageFirstIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18.41,16.59L13.82,12L18.41,7.41L17,6L11,12L17,18L18.41,16.59M6,6H8V18H6V6Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Previous Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Previous Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-75t6cl-MuiSvgIcon-root"
          data-testid="ChevronLeftIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
          />
        </svg>
      </button>
    </span>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-q2rbby-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-formControl MuiInputBase-sizeSmall css-13mvw41-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-controls=":rl:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Select"
            aria-labelledby=":rk:"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-r853lg-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id=":rk:"
            role="combobox"
            tabindex="0"
          >
            <span>
              Page 1
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
            tabindex="-1"
            value="0"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-19a9gm7-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ChevronDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-w4cd9x"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <span
      aria-label="Next Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Next Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-75t6cl-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Last Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Last Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-1doq8fr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-75t6cl-MuiSvgIcon-root"
          data-testid="PageLastIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5.59,7.41L10.18,12L5.59,16.59L7,18L13,12L7,6L5.59,7.41M16,6H18V18H16V6Z"
          />
        </svg>
      </button>
    </span>
  </div>
</div>
`;

exports[`TablePaginationActions ENCOURAGE theme matches the snapshot 1`] = `
<div>
  <div
    class="css-14q6vlk"
  >
    <span
      aria-label="First Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="First Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-bvj94t-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1mue8dp-MuiSvgIcon-root"
          data-testid="PageFirstIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18.41,16.59L13.82,12L18.41,7.41L17,6L11,12L17,18L18.41,16.59M6,6H8V18H6V6Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Previous Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Previous Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-bvj94t-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1mue8dp-MuiSvgIcon-root"
          data-testid="ChevronLeftIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
          />
        </svg>
      </button>
    </span>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-hgyret-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1v5jxjp-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-controls=":rr:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Select"
            aria-labelledby=":rq:"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-1momcb5-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id=":rq:"
            role="combobox"
            tabindex="0"
          >
            <span>
              Page 1
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
            tabindex="-1"
            value="0"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-wqvhpi-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-w4cd9x"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <span
      aria-label="Next Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Next Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-bvj94t-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1mue8dp-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Last Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Last Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-bvj94t-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1mue8dp-MuiSvgIcon-root"
          data-testid="PageLastIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5.59,7.41L10.18,12L5.59,16.59L7,18L13,12L7,6L5.59,7.41M16,6H18V18H16V6Z"
          />
        </svg>
      </button>
    </span>
  </div>
</div>
`;

exports[`TablePaginationActions ENCOURAGE_E4E theme matches the snapshot 1`] = `
<div>
  <div
    class="css-14q6vlk"
  >
    <span
      aria-label="First Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="First Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-161m8xq-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-b4t044-MuiSvgIcon-root"
          data-testid="PageFirstIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18.41,16.59L13.82,12L18.41,7.41L17,6L11,12L17,18L18.41,16.59M6,6H8V18H6V6Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Previous Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Previous Page"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-161m8xq-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-b4t044-MuiSvgIcon-root"
          data-testid="ChevronLeftIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
          />
        </svg>
      </button>
    </span>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-tn3l5o-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-h5p9cq-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-controls=":r11:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Select"
            aria-labelledby=":r10:"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-xezfm9-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r10:"
            role="combobox"
            tabindex="0"
          >
            <span>
              Page 1
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
            tabindex="-1"
            value="0"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-h5r8xo-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-w4cd9x"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <span
      aria-label="Next Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Next Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-161m8xq-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-b4t044-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
          />
        </svg>
      </button>
    </span>
    <span
      aria-label="Last Page"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="Last Page"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-161m8xq-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-b4t044-MuiSvgIcon-root"
          data-testid="PageLastIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5.59,7.41L10.18,12L5.59,16.59L7,18L13,12L7,6L5.59,7.41M16,6H18V18H16V6Z"
          />
        </svg>
      </button>
    </span>
  </div>
</div>
`;
